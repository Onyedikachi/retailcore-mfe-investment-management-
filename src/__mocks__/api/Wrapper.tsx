import { MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";

import { AppStore, RootState, store as S } from "../../config/store";
import { setupListeners } from "@reduxjs/toolkit/dist/query";
import type { PreloadedState } from "@reduxjs/toolkit";
import type { RenderOptions } from "@testing-library/react";
import { PropsWithChildren } from "react";

interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
  route?: string;
}

interface IWrapper {
  children: React.ReactNode;
}

export function renderWithProviders(
  ui: React.ReactElement | null,
  {
    preloadedState = {},
    store = S(preloadedState),
    route = "/",
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  setupListeners(store.dispatch);
  jest.mock("react-router-dom", () => ({
    BrowserRouter: ({ children }) => <div>{children}</div>,
    Routes: ({ children }) => <div>{children}</div>,
    Route: ({ element }) => element,
  }));
  function Wrappers({ children }: PropsWithChildren<IWrapper>): JSX.Element {
    window.history.pushState({}, "Test page", route);
    return <Provider store={store}>{children}</Provider>;
  }
  return { store, ...render(ui, { wrapper: Wrappers, ...renderOptions }) };
}
