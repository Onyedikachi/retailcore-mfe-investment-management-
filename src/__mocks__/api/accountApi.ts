import { axiosBaseQuery, getToken } from "@Sterling/shared";
import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
import { REHYDRATE } from "redux-persist";
import { cleanObject } from "@app/utils/cleanObject";

export const accountApi: any = createApi({
  reducerPath: "accountApi",
  baseQuery: fetchBaseQuery({
    baseUrl: "https://retailcore-accounting-api.dev.bepeerless.co/v1/",
    prepareHeaders: (headers) => {
      const token = getToken();
      if (token) {
        headers.set("authorization", `Bearer ${token}`);
      }
      return headers;
    },
  }),
  keepUnusedDataFor: 0,
  extractRehydrationInfo(action, { reducerPath }) {
    if (action.type === REHYDRATE && action.payload) {
      return action.payload[reducerPath];
    }
  },
  tagTypes: ["Get Ledgers"],
  endpoints: (builder) => ({
    getGlClass: builder.query<any, any>({
      query: () => {
        return {
          url: "accounting/glclass",
          method: "get",
        };
      },
    }),
    getLedgers: builder.query<any, any>({
      query: (data) => {
        return {
          url: `accounting/gl/leaf-ledgers/?${new URLSearchParams(
            cleanObject(data)
          )}`,
          method: "get",
        };
      },
    }),
  }),
});

export const { useGetLedgersQuery, useGetGlClassQuery } = accountApi;
