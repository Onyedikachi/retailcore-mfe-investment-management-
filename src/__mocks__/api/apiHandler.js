// mswHandlers.js
import { rest } from "msw";
import { response } from "./responses.json";

const baseUrl =
  "https://dev2-retailcore-teams-management-api.dev.bepeerless.co/v1";
const utilBaseUrl = `https://utilities-api.dev.bepeerless.co/v1`;
export const handlers = [
  rest.get(`${baseUrl}/investments`, (req, res, ctx) => {
    // Simulate different responses based on the investmentId
    if (ctx.status == 200) {
      return res(ctx.json(response.getInvestmentsResponse));
    } else {
      return res(
        ctx.status(404),
        ctx.json({
          error: "Data not found",
        })
      );
    }
  }),
  rest.get(`${baseUrl}/quick-link/all`, (req, res, ctx) => {
    // Simulate different responses based on the investmentId
    if (ctx.status == 200) {
      return res(ctx.json(response.getInvestmentsResponse));
    } else {
      return res(
        ctx.status(404),
        ctx.json({
          error: "Data not found",
        })
      );
    }
  }),
  rest.post(`${baseUrl}/investment`, (req, res, ctx) => {
    if (ctx.status == 200) {
      return res(ctx.json(response.getInvestmentsResponse));
    } else {
      return res(
        ctx.status(404),
        ctx.json({
          error: "Data not found",
        })
      );
    }
  }),
  rest.get(`${baseUrl}/investments/members/1`, (req, res, ctx) => {
    // const { filter } = req.params;
    // Simulate different responses based on the investmentId
    if (ctx.status == 200) {
      return res(ctx.json(response.getInvestmentsResponse));
    } else {
      return res(
        ctx.status(404),
        ctx.json({
          error: "Data not found",
        })
      );
    }
  }),
  rest.get(`${baseUrl}/investment/B071`, (req, res, ctx) => {
    // const { filter } = req.params;
    // Simulate different responses based on the investmentId
    if (ctx.status == 200) {
      return res(ctx.json(response.getInvestmentResponse));
    } else {
      return res(
        ctx.status(404),
        ctx.json({
          error: "Data not found",
        })
      );
    }
  }),
  rest.get(`${baseUrl}/investments/analytics`, (req, res, ctx) => {
    // const { filter } = req.params;
    // Simulate different responses based on the investmentId
    if (ctx.status == 200) {
      return res(ctx.json(response.investmentsAnalyticsResponse));
    } else {
      return res(
        ctx.status(404),
        ctx.json({
          error: "Data not found",
        })
      );
    }
  }),
  rest.get(
    `https://dev2-retailcore-teams-management-api.dev.bepeerless.co/v1/Investment/template`,
    (req, res, ctx) => {
      const mockXLSXData = `
   Name, Age
   Alice, 28
   Bob, 24
`;

      const blob = new Blob([mockXLSXData], {
        type: "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
      });
      if (ctx.status == 200) {
        return res(ctx.json({ file: blob }));
      } else {
        return res(
          ctx.status(404),
          ctx.json({
            error: "Data not found",
          })
        );
      }
    }
  ),

  rest.get(`${baseUrl}/requests`, (req, res, ctx) => {
    // Simulate different responses based on the investmentId
    if (ctx.status == 200) {
      return res(ctx.json(response.getRequestsResponse));
    } else {
      return res(
        ctx.status(404),
        ctx.json({
          error: "Data not found",
        })
      );
    }
  }),
  rest.get(`${baseUrl}/requests/edit/1`, (req, res, ctx) => {
    // Simulate different responses based on the investmentId
    if (ctx.status == 200) {
      return res(ctx.json(response.getRequestsResponse));
    } else {
      return res(
        ctx.status(404),
        ctx.json({
          error: "Data not found",
        })
      );
    }
  }),
  rest.get(`${baseUrl}/requests/edit-bulk/1`, (req, res, ctx) => {
    // Simulate different responses based on the investmentId
    if (ctx.status == 200) {
      return res(ctx.json(response.getRequestsResponse));
    } else {
      return res(
        ctx.status(404),
        ctx.json({
          error: "Data not found",
        })
      );
    }
  }),
  rest.get(`${baseUrl}/requests/analytics`, (req, res, ctx) => {
    // const { filter } = req.params;
    // Simulate different responses based on the investmentId
    if (ctx.status == 200) {
      return res(ctx.json(response.requestAnalyticsResponse));
    } else {
      return res(
        ctx.status(404),
        ctx.json({
          error: "Data not found",
        })
      );
    }
  }),
  rest.get(`${baseUrl}/approve/1`, (req, res, ctx) => {
    // const { filter } = req.params;
    // Simulate different responses based on the investmentId
    return res(ctx.json(response.requestAnalyticsResponse));
    if (ctx.status == 200) {
    } else {
      return res(
        ctx.status(404),
        ctx.json({
          error: "Data not found",
        })
      );
    }
  }),
  rest.get(`${baseUrl}/location/countries`, (req, res, ctx) => {
    // const { filter } = req.params;
    // Simulate different responses based on the investmentId
    if (ctx.status == 200) {
      return res(ctx.json(response.requestAnalyticsResponse));
    } else {
      return res(
        ctx.status(404),
        ctx.json({
          error: "Data not found",
        })
      );
    }
  }),
  rest.get(`${baseUrl}/location/lga`, (req, res, ctx) => {
    // const { filter } = req.params;
    // Simulate different responses based on the investmentId
    if (ctx.status == 200) {
      return res(ctx.json(response.requestAnalyticsResponse));
    } else {
      return res(
        ctx.status(404),
        ctx.json({
          error: "Data not found",
        })
      );
    }
  }),
  rest.post(`${baseUrl}/validate-name`, (req, res, ctx) => {
    if (ctx.status == 200) {
      return res(ctx.json({ success: true }));
    } else {
      return res(
        ctx.status(404),
        ctx.json({
          error: "Data not found",
        })
      );
    }
  }),

  // rest.get(
  //   `https://utilities-api.dev.bepeerless.co/v1/quick-link/all/ProductFactory`,
  //   (req, res, ctx) => {
  //     // const { filter } = req.params;
  //     // Simulate different responses based on the investmentId
  //     if (ctx.status == 200) {
  //       return res(ctx.json(response.requestAnalyticsResponse));
  //     } else {
  //       return res(
  //         ctx.status(500),
  //         ctx.json({
  //           status: "error",
  //           error: "Something went wrong",
  //         })
  //       );
  //     }
  //   }
  // ),


  // rest.post(
  //   `https://utilities-api.dev.bepeerless.co/v1/quick-link`,
  //   (req, res, ctx) => {
  //     // const { filter } = req.params;
  //     // Simulate different responses based on the investmentId
  //     if (ctx.status == 200) {
  //       return res(ctx.json(response.requestAnalyticsResponse));
  //     } else {
  //       return res(
  //         ctx.status(500),
  //         ctx.json({
  //           status: "error",
  //           error: "Something went wrong",
  //         })
  //       );
  //     }
  //   }
  //   ),
    rest.get(`https://retailcore-investment-management-api.dev.bepeerless.co/v1/product/product-details`,
    (req, res, ctx) => {

      if (req.id !== null ) {
        return res(ctx.json(response.getProductDetailResponseOne));
      } else {
        return res(
          ctx.status(500),
          ctx.json({
            status: "error",
            error: "Something went wrong",
          })
        );
      }
    })
    // Add more handlers for different endpoints as needed
];
