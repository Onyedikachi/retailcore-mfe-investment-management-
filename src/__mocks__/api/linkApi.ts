// @ts-nocheck
import { axiosBaseQuery, getToken } from "@Sterling/shared";
import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
import { REHYDRATE } from "redux-persist";
import { MODULENAME } from "@app/constants";
export const linkApi = createApi({
  reducerPath: "linkApi",
  baseQuery: fetchBaseQuery({
    baseUrl: `https://dev2-utilities-api.dev.bepeerless.co/v1`,
    prepareHeaders: (headers) => {
      const token = getToken();
      if (token) {
        headers.set("authorization", `Bearer ${token}`);
      }
      return headers;
    },
  }),
  extractRehydrationInfo(action, { reducerPath }) {
    if (action.type === REHYDRATE && action.payload) {
      return action.payload[reducerPath];
    }
  },
  tagTypes: ["Requests"],
  endpoints: (builder) => ({
    getLinks: builder.query<any, any>({
      query: () => {
        return {
          url: `/quick-link/all/${MODULENAME}`,
          method: "get",
        };
      },
    }),

    addLink: builder.mutation<{ id: string }, { id: string }>({
      query: (data) => {
        return {
          url: `/quick-link`,
          method: "post",
          body: data,
        };
      },
    }),
    updateLink: builder.mutation<any, any>({
      query: ({ moduleName, moduleLink }) => {
        return {
          url: `/quick-link/update/${moduleName}/${moduleLink}`,
          method: "patch",
        };
      },
    }),
    onQueryStarted() {
      // Clear previous errors when making the query
      linkApi.util.resetApiState(linkApi.endpoints.rejectRequest);
    },
  }),
});

export const { useGetLinksQuery, useAddLinkMutation, useUpdateLinkMutation } =
  linkApi;
