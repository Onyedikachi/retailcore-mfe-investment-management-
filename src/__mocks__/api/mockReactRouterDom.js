import React from "react";

const mockNavigate = jest.fn();
const mockLink = ({ to, children }) => <a href={to}>{children}</a>;

module.exports = {
  Link: mockLink,
  useNavigate: mockNavigate,
};
