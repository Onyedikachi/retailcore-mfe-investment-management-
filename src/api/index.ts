export * from "./linkApi";
export * from "./authApi";
export * from "./investmentApi";
export * from "./accountApi";
