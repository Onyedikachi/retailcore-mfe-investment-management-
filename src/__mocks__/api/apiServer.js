// mswServer.js
import { setupServer } from "msw/node";
import { handlers } from "./apiHandler";

// Create the MSW server instance
export const server = setupServer(...handlers);

// Export the server instance for use in tests
// export { server };
