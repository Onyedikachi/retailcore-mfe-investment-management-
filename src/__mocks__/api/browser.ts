// mswServer.js
import { setupWorker } from "msw";
import { handlers } from "./apiHandler";

// Create the MSW server instance
export const server = setupWorker(...handlers);

// Export the server instance for use in tests
// export { server };
