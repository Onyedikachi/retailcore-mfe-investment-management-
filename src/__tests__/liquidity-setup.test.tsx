import { render, screen } from "@testing-library/react";
import LiquiditySetup from "../components/pages/term-deposit/forms/liquidity-setup"
import {renderWithProviders} from "../__mocks__/api/Wrapper"

const fData = {
    part_AllowPartLiquidation: false,
    part_MaxPartLiquidation: "",
    part_RequireNoticeBeforeLiquidation: false,
    part_NoticePeriod: "",
    part_NoticePeriodUnit: "",
    part_LiquidationPenalty: "",
    part_LiquidationPenaltyPercentage: "",
    early_AllowEarlyLiquidation: "",
    early_RequireNoticeBeforeLiquidation: false,
    early_NoticePeriod: "",
    early_NoticePeriodUnit: "",
    early_LiquidationPenalty: "",
    early_LiquidationPenaltyPercentage: ""
}

describe("LiquiditySetup", () => {
    it ("Renders without error", () => {
        const form = renderWithProviders(<LiquiditySetup proceed={jest.fn()} formData={fData} setDisabled={jest.fn()}/>)
        expect(form).toMatchSnapshot()
    })
})