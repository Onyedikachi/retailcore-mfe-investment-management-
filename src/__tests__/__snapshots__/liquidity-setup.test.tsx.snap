// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LiquiditySetup Renders without error 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        id="liquiditysetup"
      >
        <div
          class="flex flex-col gap-14"
        >
          <div
            class="rounded-[5px] px-10 flex flex-col gap-[40px] py-5"
            style="box-shadow: 0px 0px 1px 0px rgba(26, 32, 36, 0.32), 0px 1px 2px 0px rgba(91, 104, 113, 0.32);"
          >
            <div
              class="flex items-center gap-[10px]"
            >
              <span
                class="capitalize min-w-[300px] flex items-center gap-[5px] text-[##636363] text-base font-medium"
              >
                Allow Part Liquidation
                <div
                  class="flex items-end cursor-pointer"
                >
                  <div
                    class="flex items-end !bg-white"
                  >
                    <button
                      aria-expanded="false"
                      class="bg-transparent p-0"
                      type="button"
                    >
                      <div
                        class="w-[18px] h-[18px] text-[#636363]"
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          version="1.1"
                          viewBox="0 0 16 16"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M7 4.75c0-0.412 0.338-0.75 0.75-0.75h0.5c0.412 0 0.75 0.338 0.75 0.75v0.5c0 0.412-0.338 0.75-0.75 0.75h-0.5c-0.412 0-0.75-0.338-0.75-0.75v-0.5z"
                          />
                          <path
                            d="M10 12h-4v-1h1v-3h-1v-1h3v4h1z"
                          />
                          <path
                            d="M8 0c-4.418 0-8 3.582-8 8s3.582 8 8 8 8-3.582 8-8-3.582-8-8-8zM8 14.5c-3.59 0-6.5-2.91-6.5-6.5s2.91-6.5 6.5-6.5 6.5 2.91 6.5 6.5-2.91 6.5-6.5 6.5z"
                          />
                        </svg>
                      </div>
                    </button>
                  </div>
                </div>
              </span>
              <button
                aria-checked="false"
                class="bg-transparent border-[#CF2A2A] relative inline-flex h-4 w-7 flex-shrink-0 cursor-pointer rounded-full border  transition-colors duration-200 ease-in-out focus:outline-none ring-0  "
                data-headlessui-state=""
                id="headlessui-switch-1"
                role="switch"
                tabindex="0"
                type="button"
                value="false"
              >
                <span
                  class="sr-only"
                >
                  Use setting
                </span>
                <span
                  aria-hidden="true"
                  class="translate-x-0  bg-white  pointer-events-none inline-block h-[14px] w-[14px] transform rounded-full border border-[#CF2A2A] shadow ring-0 transition duration-200 ease-in-out"
                />
              </button>
            </div>
          </div>
          <div
            class="rounded-[5px] px-10 flex flex-col gap-[40px] py-5"
            style="box-shadow: 0px 0px 1px 0px rgba(26, 32, 36, 0.32), 0px 1px 2px 0px rgba(91, 104, 113, 0.32);"
          >
            <div
              class="flex items-center gap-[10px]"
            >
              <span
                class="capitalize min-w-[300px] flex items-center gap-[5px] text-[##636363] text-base font-medium"
              >
                Allow Early Liquidation
                <div
                  class="flex items-end cursor-pointer"
                >
                  <div
                    class="flex items-end !bg-white"
                  >
                    <button
                      aria-expanded="false"
                      class="bg-transparent p-0"
                      type="button"
                    >
                      <div
                        class="w-[18px] h-[18px] text-[#636363]"
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          version="1.1"
                          viewBox="0 0 16 16"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M7 4.75c0-0.412 0.338-0.75 0.75-0.75h0.5c0.412 0 0.75 0.338 0.75 0.75v0.5c0 0.412-0.338 0.75-0.75 0.75h-0.5c-0.412 0-0.75-0.338-0.75-0.75v-0.5z"
                          />
                          <path
                            d="M10 12h-4v-1h1v-3h-1v-1h3v4h1z"
                          />
                          <path
                            d="M8 0c-4.418 0-8 3.582-8 8s3.582 8 8 8 8-3.582 8-8-3.582-8-8-8zM8 14.5c-3.59 0-6.5-2.91-6.5-6.5s2.91-6.5 6.5-6.5 6.5 2.91 6.5 6.5-2.91 6.5-6.5 6.5z"
                          />
                        </svg>
                      </div>
                    </button>
                  </div>
                </div>
              </span>
              
              <button
                aria-checked=""
                class="bg-transparent border-[#CF2A2A] relative inline-flex h-4 w-7 flex-shrink-0 cursor-pointer rounded-full border  transition-colors duration-200 ease-in-out focus:outline-none ring-0  "
                id="headlessui-switch-2"
                role="switch"
                tabindex="0"
                type="button"
                value=""
              >
                <span
                  class="sr-only"
                >
                  Use setting
                </span>
                <span
                  aria-hidden="true"
                  class="translate-x-0  bg-white  pointer-events-none inline-block h-[14px] w-[14px] transform rounded-full border border-[#CF2A2A] shadow ring-0 transition duration-200 ease-in-out"
                />
              </button>
            </div>
            
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      id="liquiditysetup"
    >
      <div
        class="flex flex-col gap-14"
      >
        <div
          class="rounded-[5px] px-10 flex flex-col gap-[40px] py-5"
          style="box-shadow: 0px 0px 1px 0px rgba(26, 32, 36, 0.32), 0px 1px 2px 0px rgba(91, 104, 113, 0.32);"
        >
          <div
            class="flex items-center gap-[10px]"
          >
            <span
              class="capitalize min-w-[300px] flex items-center gap-[5px] text-[##636363] text-base font-medium"
            >
              Allow Part Liquidation
              <div
                class="flex items-end cursor-pointer"
              >
                <div
                  class="flex items-end !bg-white"
                >
                  <button
                    aria-expanded="false"
                    class="bg-transparent p-0"
                    type="button"
                  >
                    <div
                      class="w-[18px] h-[18px] text-[#636363]"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        version="1.1"
                        viewBox="0 0 16 16"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M7 4.75c0-0.412 0.338-0.75 0.75-0.75h0.5c0.412 0 0.75 0.338 0.75 0.75v0.5c0 0.412-0.338 0.75-0.75 0.75h-0.5c-0.412 0-0.75-0.338-0.75-0.75v-0.5z"
                        />
                        <path
                          d="M10 12h-4v-1h1v-3h-1v-1h3v4h1z"
                        />
                        <path
                          d="M8 0c-4.418 0-8 3.582-8 8s3.582 8 8 8 8-3.582 8-8-3.582-8-8-8zM8 14.5c-3.59 0-6.5-2.91-6.5-6.5s2.91-6.5 6.5-6.5 6.5 2.91 6.5 6.5-2.91 6.5-6.5 6.5z"
                        />
                      </svg>
                    </div>
                  </button>
                </div>
              </div>
            </span>
            <button
              aria-checked="false"
              class="bg-transparent border-[#CF2A2A] relative inline-flex h-4 w-7 flex-shrink-0 cursor-pointer rounded-full border  transition-colors duration-200 ease-in-out focus:outline-none ring-0  "
              data-headlessui-state=""
              id="headlessui-switch-1"
              role="switch"
              tabindex="0"
              type="button"
              value="false"
            >
              <span
                class="sr-only"
              >
                Use setting
              </span>
              <span
                aria-hidden="true"
                class="translate-x-0  bg-white  pointer-events-none inline-block h-[14px] w-[14px] transform rounded-full border border-[#CF2A2A] shadow ring-0 transition duration-200 ease-in-out"
              />
            </button>
          </div>
        </div>
        <div
          class="rounded-[5px] px-10 flex flex-col gap-[40px] py-5"
          style="box-shadow: 0px 0px 1px 0px rgba(26, 32, 36, 0.32), 0px 1px 2px 0px rgba(91, 104, 113, 0.32);"
        >
          <div
            class="flex items-center gap-[10px]"
          >
            <span
              class="capitalize min-w-[300px] flex items-center gap-[5px] text-[##636363] text-base font-medium"
            >
              Allow Early Liquidation
              <div
                class="flex items-end cursor-pointer"
              >
                <div
                  class="flex items-end !bg-white"
                >
                  <button
                    aria-expanded="false"
                    class="bg-transparent p-0"
                    type="button"
                  >
                    <div
                      class="w-[18px] h-[18px] text-[#636363]"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        version="1.1"
                        viewBox="0 0 16 16"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M7 4.75c0-0.412 0.338-0.75 0.75-0.75h0.5c0.412 0 0.75 0.338 0.75 0.75v0.5c0 0.412-0.338 0.75-0.75 0.75h-0.5c-0.412 0-0.75-0.338-0.75-0.75v-0.5z"
                        />
                        <path
                          d="M10 12h-4v-1h1v-3h-1v-1h3v4h1z"
                        />
                        <path
                          d="M8 0c-4.418 0-8 3.582-8 8s3.582 8 8 8 8-3.582 8-8-3.582-8-8-8zM8 14.5c-3.59 0-6.5-2.91-6.5-6.5s2.91-6.5 6.5-6.5 6.5 2.91 6.5 6.5-2.91 6.5-6.5 6.5z"
                        />
                      </svg>
                    </div>
                  </button>
                </div>
              </div>
            </span>
            
            <button
              aria-checked=""
              class="bg-transparent border-[#CF2A2A] relative inline-flex h-4 w-7 flex-shrink-0 cursor-pointer rounded-full border  transition-colors duration-200 ease-in-out focus:outline-none ring-0  "
              id="headlessui-switch-2"
              role="switch"
              tabindex="0"
              type="button"
              value=""
            >
              <span
                class="sr-only"
              >
                Use setting
              </span>
              <span
                aria-hidden="true"
                class="translate-x-0  bg-white  pointer-events-none inline-block h-[14px] w-[14px] transform rounded-full border border-[#CF2A2A] shadow ring-0 transition duration-200 ease-in-out"
              />
            </button>
          </div>
          
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
